PyEMD/EMD.py

```diff
@@ -748,7 +748,7 @@ def emd(self, S, T=None, max_imf=None):
             Set of IMFs produced from input signal.
         """
 
-        if T is None: T = np.arange(len(S), dtype=S.dtype)

```

Comment:
This case is mainly for unevenly sampled data for which we need to use different than "simple" extrema detection. An example of such case is with `t = [0, 0.5, 1.5, 2]` and `s = [0, 0.5, 0.5, 0]` for which a better guess of a maximum is `(t, s) = (1, 1)` which is not in the input array. Such case is handled when `self.extrema_detection != "simple"`.

I like the idea of forcing indexed based array for a simple case, so let's maybe refactor this to:
```python
if T is None or self.extrema_detection == "simple":
    T = np.arange(len(s), dtype=S.dtype)
```

Reply:
OK, I see your point and have made the changes